src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -93,5 +97,96 @@ public PagedDebtPositions getPagedDebtPositionsByIngestionFlowFileId(Long ingest
   public void delete(DebtPosition debtPosition) {
     debtPositionDeleteService.delete(debtPosition);
   }
+
+  @Override
+  public Optional<DebtPosition> getDebtPositionByIupdAndOrganizationId(String iupd, Long organizationId) {
+    return debtPositionRepository.findDebtPositionByIupdOrgAndOrganizationId(iupd, organizationId);
+  }
+
+  @Override
+  public void updateDebtPosition(Long debtPositionId, DebtPositionDTO debtPositionDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findEntityGraphByDebtPositionId(debtPositionId);
+    if (debtPosition == null) {
+      throw new EntityNotFoundException("DebtPosition with id %d not found".formatted(debtPositionId));
+    }
+
+    propagateIdsForDebtPosition(debtPosition, debtPositionDTO);
+
+    debtPositionSaveService.saveDebtPositionDTO(debtPositionDTO);
+  }
+
+  private void propagateIdsForDebtPosition(DebtPosition entity, DebtPositionDTO dto) {
+    if (entity == null || dto == null) return;
+
+    if (dto.getDebtPositionId() == null && entity.getDebtPositionId() != null) {
+      dto.setDebtPositionId(entity.getDebtPositionId());
+    }
```

Comment:
we should always override the id, and entity id cannot be null
```suggestion
    dto.setDebtPositionId(entity.getDebtPositionId());
```